applications/infra.py

```diff
@@ -0,0 +1,241 @@
+# E.g. run with
+# export PYTHONPATH=${PYTHONPATH}:.
+# python applications/infra.py
+# or (macos):
+# pythonw applications/infra.py
+
+import matplotlib.pyplot as plt
+import nevergrad as ng
+import numpy as np
+import random
+import math
+pi = math.pi
+
+# We consider a power systems problem.
+# We have N stocks.
+# Each stock is managed by an agent.
+# Agents collaborate for minimizing the total cost.
+
+# Number of stocks (dams).
+N = 7
+
+# Optimization budget.
+optim_steps = 8
+
+# Real life is more complicated! This is a very simple model.
+
+class Agent():
+    """An agent has an input size, an output size, a number of layers, a width of its internal layers 
+    (a.k.a number of neurons per hidden layer)."""
+    def __init__(self, input_size, output_size, layers=2, layer_width=20):
+        assert(layers >= 2)
+        self.input_size = input_size
+        self.output_size = output_size
+        self.layers = []
+        self.layers += [np.random.rand(int(input_size), int(layer_width))]
+        for i in range(layers-2):
+            self.layers += [np.random.rand(int(layer_width), int(layer_width))]
```

Comment:
I was thinking that maybe you should initialize your weights with Xavier instead of something uniform. I guess it would also work better with some of the genetic algorithms.

Reply:
Done + some related modifications.